packages/form-upload/tests/filesDelivery.test.js

```diff
@@ -0,0 +1,227 @@
+      )
+      .reply(
+        204,
+        {},
+        {
+          'tus-resumable': '1.0.0',
+          'upload-expires': 'Fri, 12 Jan 2030 15:54:39 GMT',
+          'transfer-encoding': 'chunked',
+          'Upload-Offset': 12,
+          references: '["/files/105265/9ee77f6d-9779-4b96-a995-0df47657e504"]',
+        }
+      );
+
+    xhrMock.use('HEAD', /.*4611142db7c049bbbe37376583a3f46b.*/, {
+      status: 200,
+      headers: {
+        'Content-Length': '0',
+        'AV-Scan-Result': 'accepted',
+        'Upload-Result': 'accepted',
+      },
+    });
+  });
+
+  afterEach(() => {
+    xhrMock.teardown();
+  });
+
+  test('calls avFilesDeliveryApi when deliveryChannel and fileDeliveryMetadata are defined and deliverFileOnSubmit is false', async done => {
+    avFilesDeliveryApi.uploadFilesDelivery.mockResolvedValue({
+      id: '123456',
+      status: 'COMPLETE',
+    });
+
+    const { getByTestId } = renderUpload(
+      {
+        initialValues: {
+          upload: null,
+        },
+      },
+      {
+        name: 'upload',
+        clientId: 'a',
+        bucketId: 'testbucket',
+        customerId: 'c',
+        deliveryChannel: 'test',
+        deliverFileOnSubmit: false,
+        fileDeliveryMetadata: {
+          payerId: 'AvailityTest',
+        },
+      }
+    );
+
+    const file = Buffer.from('hello world!');
+    file.name = 'a';
+    const fileEvent = {
+      target: {
+        files: [file],
+      },
+    };
+
+    const upload = createTestExtendingUpload(file, done);
+    upload.start();
+
+    const inputNode = getByTestId('file-picker');
+
+    fireEvent.change(inputNode, fileEvent);
+    expect(inputNode.files.length).toBe(1);
+
+    await wait(() =>
+      expect(avFilesDeliveryApi.uploadFilesDelivery).toHaveBeenCalled()
+    );
+  });
+
+  test('calls avFilesDeliveryApi onSubmit when deliverFileOnSubmit is true and deliveryChannel and fileDeliveryMetadata are defined', async done => {
+    const onSubmit = jest.fn();
+    avFilesDeliveryApi.uploadFilesDelivery.mockResolvedValue({
+      id: '123456',
+      status: 'COMPLETE',
+    });
+
+    const { getByTestId, getByText } = renderUpload(
+      { initialValues: { upload: null }, onSubmit },
+      {
+        name: 'upload',
+        clientId: 'a',
+        bucketId: 'testbucket',
+        customerId: 'c',
+        deliverFileOnSubmit: true,
+        deliverChannel: 'test',
+        fileDeliveryMetadata: { payerId: 'AvailityTest' },
+      }
+    );
+
+    const inputNode = getByTestId('file-picker');
+
+    const file = Buffer.from('hello world!');
+    file.name = 'a';
+    const fileEvent = {
+      target: {
+        files: [file],
+      },
+    };
+
+    const upload = createTestExtendingUpload(file, done);
+    upload.start();
+
+    fireEvent.change(inputNode, fileEvent);
+    expect(inputNode.files.length).toBe(1);
+
+    fireEvent.click(getByText('Submit'));
+    await wait(() => expect(onSubmit).toHaveBeenCalled());
```

Comment:
`wait` has been deprecated in favor of `waitFor` in a future version of RTL. Let's future proof this a bit

Reply:
Just documenting that we saw some breaking changes when switching to `waitFor` and agreed to investigate/address this in a follow up PR or an addition to https://github.com/Availity/availity-react/pull/631